src/serializers/type_serializers/union.rs

```diff
@@ -169,8 +190,14 @@ fn serde_serialize<S: serde::ser::Serializer>(
         }
     }
 
-    for err in &errors {
-        extra.warnings.custom_warning(err.to_string());
+    // If extra.check is SerCheck::None, we're in a top-level union. We should thus raise the warnings
+    if extra.check == SerCheck::None {
+        for err in &errors {
+            extra.warnings.custom_warning(err.to_string());
+        }
+    } else {
+        // NOTE: if this function becomes recursive at some point, an `Err(_)` containing the errors
+        // will have to be returned here
```

Comment:
I've left this empty for now, as we can't return a `PydanticSerializationUnexpectedValue` here since the function is supposed to return `S::Error`., and refactoring this would introduce extra complexity